spec/app/domain/commands/credentials/rotate_api_key_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'spec_helper'
+
+describe Commands::Credentials::RotateApiKey do
+  let(:credentials) { double(Credentials) }
+  let(:role) { double(Role, id: 'role', credentials: credentials) }
+
+  let(:other_credentials) { double(Credentials) }
+  let(:other_role) { double(Role, id: 'other role', credentials: other_credentials) }
+
+  let(:role_to_rotate) { role }
+
+  let(:err_message) { 'the error message' }
+
+  let(:audit_log) { double(::Audit.logger) }
+
+  let(:audit_success) do
+    ::Audit::Event::ApiKey.new(
+      auth_role_id: role.id,
+      subject_role_id: role_to_rotate.id,
+      success: true
+    )
+  end
+
+  let(:audit_failure) do
+    ::Audit::Event::ApiKey.new(
+      auth_role_id: role.id,
+      subject_role_id: role_to_rotate.id,
+      success: false,
+      error_message: err_message
+    )
+  end
+
+  subject do 
+    Commands::Credentials::RotateApiKey.new(
+      audit_log: audit_log
+    )
+  end
+
+  context 'when rotating own API key' do
+    it 'updates the key' do
+      # Expect it to rotate the api key on the credentials model, and to save it
+      expect(credentials).to receive(:rotate_api_key)
+      expect(credentials).to receive(:save)
```

Comment:
Per the very first part of our convo....

Since normally it's incorrect to use an input _mock_ (those would typically be doubles, and only the deps would be mocks with receive assertions), let's add a comment here that we're only breaking the rule because the correct fix (refactoring the entire credentials controller) was out of scope.

Reply:
Ah, that's right. Will do.